src/MoBi.Core/Commands/ChangeValueFormulaCommand.cs

```diff
@@ -63,21 +61,54 @@ protected override void ExecuteWith(IMoBiContext context)
       {
          base.ExecuteWith(context);
          _changedStartValue.Formula = _newFormula;
-         Path = _changedStartValue.Path;
+         Path = PathFrom(_changedStartValue);
       }
 
-      protected override ICommand<IMoBiContext> GetInverseCommand(IMoBiContext context)
-      {
-         return new ChangeStartValueFormulaCommand<T>(_startValuesBuildingBlock, _changedStartValue, _oldFormula, _newFormula).AsInverseFor(this);
-      }
+      protected abstract IObjectPath PathFrom(T changedStartValue);
+
 
       public override void RestoreExecutionData(IMoBiContext context)
       {
          base.RestoreExecutionData(context);
-         _startValuesBuildingBlock = context.Get<IStartValuesBuildingBlock<T>>(_objectBaseId);
-         _changedStartValue = _startValuesBuildingBlock.Single(startValue => startValue.Path.Equals(Path));
+         _buildingBlock = context.Get<IBuildingBlock<T>>(_objectBaseId);
+         _changedStartValue = _buildingBlock.Single(startValue => PathFrom(startValue).Equals(Path));
          _oldFormula = GetFormula(_oldFormulaId, context);
          _newFormula = GetFormula(_newFormulaId, context);
       }
    }
+
+   public class ExpressionParameterFormulaChangedCommand : ChangeValueFormulaCommand<ExpressionParameter>
```

Comment:
I don't understand why we need different commands here. It looks like the exact same to me. Just make sure they all have a common interface? What am I missing?

Reply:
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/issues/1811